tests/_test_tools.py

```diff
@@ -53,3 +57,17 @@ def _get_executor_instance(ep: ExecutorTestParams, job: Optional[Job] = None) ->
         if ep.queue_name is not None:
             job.spec.attributes.queue_name = ep.queue_name
     return JobExecutor.get_instance(ep.executor, url=ep.url)
+
+
+@contextmanager
+def _deploy(path: Union[Path, str]) -> Iterator[Path]:
+    # Copies `path` to a directory assumed to be on a shared FS (~/.psij/test) and
+    # returns the resulting path
+    if isinstance(path, str):
+        path = Path(path)
+    with tempfile.NamedTemporaryFile(dir=Path.home() / '.psij' / 'test', delete=False) as df:
```

Comment:
Why the `delete=False` and `finally` instead of `delete=True` (which is the default anyway):
```py
with tempfile.NamedTemporaryFile(dir=Path.home() / '.psij' / 'test') as df:
    shutil.copyfile(path, df.name)
    yield Path(df.name)
```

Reply:
Good point and thanks for commenting on this, since I forgot to close the file. With `delete=True` it deletes on close. So if the scenario is to manipulate the file outside of python, you'd close it and pass the name around instead. Without `delete=False`, the file will be deleted as soon as it is closed. The need to close is because on some platforms (Windows) the file cannot be opened concurrently. The question that the Python documentation doesn't answer is whether, if the file is deleted, the name property would still be valid. The wording of the documentation suggests that this may not be the case.

Reply:
Yeah, makes sense now, thanks!